core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1755,6 +1758,7 @@ class KafkaController(val config: KafkaConfig,
     }
 
     if (!isActive) return
+    rearrangePartitionReplicaAssignmentForNewPartitions(immutable.Set(topic))
```

Comment:
does it conflict with partition reassignment?
say if a partition gets reassigned to replica ( 1, 2, 5, 6 ) ==> if one of this replica is maintenance brokers, would this reassignment complete if we automatically changing zk node to disallow placing replicas on maintenance brokers.

Reply:
That's a good point. I've updated the PR to avoid assignment replicas to the undesirable hosts during partition reassignment. Please take another look. Thanks!